managed/services/grafana/client.go

```diff
@@ -214,27 +217,44 @@ func (c *Client) GetUserID(ctx context.Context) (int, error) {
 	return int(userID), nil
 }
 
+var emptyUser = authUser{
+	role:   none,
+	userID: 0,
+}
+
 // getAuthUser returns grafanaAdmin if currently authenticated user is a Grafana (super) admin.
 // Otherwise, it returns a role in the default organization (with ID 1).
 // Ctx is used only for cancelation.
 func (c *Client) getAuthUser(ctx context.Context, authHeaders http.Header) (authUser, error) {
-	// Check if it's API Key
-	if c.isAPIKeyAuth(authHeaders.Get("Authorization")) {
+	// Check if API Key or Service Token is authorized.
+	token := c.getTokenAuth(authHeaders)
+	if token != "" {
+		if strings.HasPrefix(token, "glsa_") {
```

Comment:
service accounts might work with usual API keys as well. What if we migrate the whole logic about getting role from bearer from pmm-managed to grafana since we know if it's API Key or service token on grafana side?